packages/plugin-eslint/src/lib/meta/rules.spec.ts

```diff
@@ -0,0 +1,241 @@
+            {
+              additionalProperties: false,
+              properties: {
+                checkFragmentShorthand: { default: false, type: 'boolean' },
+                checkKeyMustBeforeSpread: { default: false, type: 'boolean' },
+                warnOnDuplicates: { default: false, type: 'boolean' },
+              },
+              type: 'object',
+            },
+          ],
+        },
+        options: [],
+      } satisfies RuleData);
+    });
+  });
+
+  describe('Nx monorepo', () => {
+    const nxRootDir = join(fixturesDir, 'nx-monorepo');
+    const eslintrc = join(nxRootDir, 'packages/utils/.eslintrc.json');
+
+    const eslint = new ESLint({
+      useEslintrc: false,
+      baseConfig: { extends: eslintrc },
+    });
+    const patterns = ['packages/utils/**/*.ts', 'packages/utils/**/*.json'];
+
+    beforeAll(() => {
+      cwdSpy.mockReturnValue(nxRootDir);
+    });
+
+    it('should list expected number of rules', async () => {
+      await expect(listRules(eslint, patterns)).resolves.toHaveLength(69);
+    });
+
+    it('should include explicitly set plugin rule with custom options', async () => {
+      await expect(listRules(eslint, patterns)).resolves.toContainEqual({
+        ruleId: '@nx/enforce-module-boundaries',
+        meta: {
+          docs: {
+            description:
+              'Ensure that module boundaries are respected within the monorepo',
+            // @ts-expect-error Nx rule produces string, but ESLint types expect boolean
+            recommended: 'error',
+            url: '',
+          },
+          fixable: 'code',
+          messages: expect.any(Object),
+          schema: [
+            {
+              additionalProperties: false,
+              properties: expect.any(Object),
+              type: 'object',
+            },
+          ],
+          type: 'suggestion',
+        },
+        options: [
+          {
+            allow: [],
+            depConstraints: [
+              {
+                onlyDependOnLibsWithTags: ['*'],
+                sourceTag: '*',
+              },
+            ],
+            enforceBuildableLibDependency: true,
+          },
+        ],
+      } satisfies RuleData);
+    });
+
+    it('should include built-in rule set implicitly by extending recommended config', async () => {
+      await expect(listRules(eslint, patterns)).resolves.toContainEqual({
+        ruleId: 'no-var',
+        meta: {
+          docs: {
+            description: 'Require `let` or `const` instead of `var`',
+            recommended: false,
+            url: 'https://eslint.org/docs/latest/rules/no-var',
+          },
+          fixable: 'code',
+          messages: {
+            unexpectedVar: 'Unexpected var, use let or const instead.',
+          },
+          schema: [],
+          type: 'suggestion',
+        },
+        options: [],
+      } as RuleData);
+    });
+
+    it('should include plugin rule set implicitly by extending recommended config', async () => {
```

Comment:
It would be valuable to add a note in regards to which config that is supposed to be for future reference. Also related to the other implicit rule.

Reply:
Added some comments.